pkg/kubelet/kubelet.go

```diff
@@ -1541,7 +1541,7 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if !runnable.Admit || pod.DeletionTimestamp != nil || apiPodStatus.Phase == v1.PodFailed {
+	if !runnable.Admit || pod.DeletionTimestamp != nil || apiPodStatus.Phase == v1.PodFailed || apiPodStatus.Phase == v1.PodSucceeded {
```

Comment:
If we delete succeeded pod like this, will users miss the SUCCESS status of Job during kubectl get?